src/hotspot/share/runtime/arguments.cpp

```diff
@@ -1993,7 +1993,9 @@ bool Arguments::check_vm_args_consistency() {
                 "-XX:+VerifyHeavyMonitors requires -XX:+UseHeavyMonitors");
     return false;
   }
-
+  if (UseHeavyMonitors) {
+    FLAG_SET_DEFAULT(UseFastLocking, false);
```

Comment:
Probably should be an error if both are set on the command-line.

Reply:
I am not sure. UseFastLocking (or whatever we will call it in the near future) chooses between traditional and new stack-locking. We disable both by +UseHeavyMonitors. Also, +UseFastLocking is not really meant to be used by users, at least it is my intention to mainly turn it on programatically it when compact headers (Lilliput) is turned on. When also running with +UseHeavyMonitors, then it doesn't really matter which stack-locking impl is selected, neither would actually be used. I'd rather remove the explicit turning-off of UseFastMonitors under +UseHeavyMonitors.